hoomd/mpcd/DryingDropletStreaming.h

```diff
@@ -0,0 +1,113 @@
+ *  3. validateBox(): Checks whether the global simulation box is consistent with the streaming geometry.
+ *
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param geom Streaming geometry
+         * \param R is the radius of sphere
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase, std::shared_ptr<::Variant> R, boundary bc)
+        : mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::SphereGeometry>()),
+          m_R(R),m_bc(bc)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+    private:
+        std::shared_ptr<::Variant> m_R; //!Radius of Sphere
+        const boundary m_bc;            //!boundary conditions
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod<mpcd::SphereGeometry>::stream(unsigned int timestep)
+    {
+    const double start_R = (*m_R)(timestep);
+    const double end_R = (*m_R)(timestep + m_period);
+    const double V = (end_R - start_R)/(m_period * m_mpcd_dt);
+    if (V>0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+    m_validate_geom = m_geom; //validating the geometry
+    m_geom = std::make_shared<const mpcd::SphereGeometry>(end_R, V, m_bc);
+    //stream using ConfinedStreamingMethod
+    ConfinedStreamingMethod<mpcd::SphereGeometry>::stream(timestep);
+    //delete marked particles
+    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::overwrite);
+    
+    }
+
+
+namespace detail
+{
+//! Export mpcd::DryingDropletStreamingMethod to python
+/*!
+ * \param m Python module to export to
+ */
+
+void export_DryingDropletStreamingMethod(pybind11::module& m)
+    {
+    namespace py = pybind11;
+    const std::string name = "ConfinedStreamingMethod" + Geometry::getName();
+    py::class_<mpcd::DryingDropletStreamingMethod<mpcd::SphereGeometry>, std::shared_ptr<DryingDropletStreamingMethod<mpcd::SphereGeometry>>>
+        (m, name.c_str(), py::base<mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>>())
+        .def(py::init<std::shared_ptr<mpcd::SystemData>, unsigned int, unsigned int, int, std::shared_ptr<::Variant>, boundary>())
+        .def_property("Spheregeometry" ,&mpcd::DryingDropletStreamingMethod<mpcd::SphereGeometry>::setGeometry);
```

Comment:
I would guess this does not actually compile? I think it should be like this:
```suggestion
    py::class_<mpcd::DryingDropletStreamingMethod, mpcd::ConfinedStreamingMethod<mpcd::SphereGeometry>, std::shared_ptr<DryingDropletStreamingMethod>>
        (m, "DryingDropletStreamingMethod")
        .def(py::init<std::shared_ptr<mpcd::SystemData>, unsigned int, unsigned int, int, std::shared_ptr<::Variant>, boundary>());
```
so that it inherits from the right base class